Microsoft.Toolkit.Uwp.UI/Cache/CacheBase.cs

```diff
@@ -414,23 +509,35 @@ private static ulong CreateHash64(string str)
         {
             T instance = default(T);
 
-            using (var webStream = await StreamHelper.GetHttpStreamAsync(uri, cancellationToken).ConfigureAwait(MaintainContext))
+            using (InMemoryRandomAccessStream randomAccessStream = new InMemoryRandomAccessStream())
             {
+                var dataStream = randomAccessStream.AsStreamForWrite();
+
+                using (var response = await HttpClient.GetAsync(uri, cancellationToken))
+                {
+                    using (var stream = await response.Content.ReadAsStreamAsync())
+                    {
+                        stream.CopyTo(dataStream);
+                        stream.Position = 0;
+
+                        using (var fs = await baseFile.OpenStreamForWriteAsync())
+                        {
+                            stream.CopyTo(fs);
+
+                            fs.Flush();
+                        }
+                    }
+                }
+
                 // if its pre-cache we aren't looking to load items in memory
                 if (!preCacheOnly)
                 {
-                    instance = await InitializeTypeAsync(webStream, initializerKeyValues).ConfigureAwait(false);
+                    randomAccessStream.Seek(0);
 
-                    webStream.Seek(0);
+                    instance = await InitializeTypeAsync(randomAccessStream, initializerKeyValues).ConfigureAwait(false);
```

Comment:
Bitmap requests IRandomAccessStream. The original stream is Stream and the operation has to be done somewhere.